bulk-model-sync-gradle/src/main/kotlin/org/modelix/model/sync/bulk/gradle/tasks/ExportFromModelServer.kt

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2023.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.modelix.model.sync.bulk.gradle.tasks
+
+import kotlinx.coroutines.runBlocking
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.TaskAction
+import org.modelix.model.ModelFacade
+import org.modelix.model.api.IBranch
+import org.modelix.model.api.IProperty
+import org.modelix.model.api.PBranch
+import org.modelix.model.api.getRootNode
+import org.modelix.model.client2.IModelClientV2
+import org.modelix.model.client2.ModelClientV2
+import org.modelix.model.client2.ModelClientV2PlatformSpecificBuilder
+import org.modelix.model.client2.getReplicatedModel
+import org.modelix.model.data.NodeData
+import org.modelix.model.lazy.RepositoryId
+import org.modelix.model.mpsadapters.RepositoryLanguage
+import org.modelix.model.sync.bulk.ModelExporter
+import javax.inject.Inject
+
+abstract class ExportFromModelServer @Inject constructor(of: ObjectFactory) : DefaultTask() {
+
+    @Input
+    val url: Property<String> = of.property(String::class.java)
+
+    @Input
+    @Optional
+    val repositoryId: Property<String> = of.property(String::class.java)
+
+    @Input
+    @Optional
+    val branchName: Property<String> = of.property(String::class.java)
+
+    @Input
+    @Optional
+    val revision: Property<String> = of.property(String::class.java)
+
+    @OutputDirectory
+    val outputDir: DirectoryProperty = of.directoryProperty()
+
+    @TaskAction
+    fun export() {
```

Comment:
Why isn't the whole method being executed in a big `runBlocking` statement? Right now, these ugly things are somewhere in every detail method.

Reply:
Raising the indent level for everything just because of one coroutine call sounds pretty ugly to me. But I will improve the readability for this case in another way.

Reply:
Seems to be tha multiple coroutines are called? Otherwise there wouldn't be the need for multiple runBlockings in called methods?